src/auth/token-verifier.ts

```diff
@@ -397,9 +506,12 @@ export class FirebaseTokenVerifier {
     } else if (!isEmulator && header.alg !== ALGORITHM_RS256) {
       errorMessage = `${this.tokenInfo.jwtName} has incorrect algorithm. Expected "` + ALGORITHM_RS256 + '" but got ' +
         '"' + header.alg + '".' + verifyJwtTokenDocsMessage;
-    } else if (payload.aud !== projectId) {
+    } else if (audience && !(payload.aud as string).includes(audience)) {
```

Comment:
I think it is better to check the type here
```
if (typeof audience !== 'undefined' && ...
```

Reply:
SGTM